app/controllers/concerns/wrap_in_with_deleted.rb

```diff
@@ -3,29 +3,21 @@ module WrapInWithDeleted
   extend ActiveSupport::Concern
 
   included do
-    around_action :wrap_in_with_deleted, if: proc {
-      request.get? && (with_deleted? || search_with_deleted?)
-    }
+    around_action :wrap_in_with_deleted
   end
 
   protected
 
-  def wrap_in_with_deleted
-    Project.with_deleted do
-      Stage.with_deleted do
-        Deploy.with_deleted do
-          yield
-        end
-      end
+  def wrap_in_with_deleted(&block)
+    if request.get? && value = (params[:with_deleted].presence || search_with_deleted?)
+      klasses = value.split(",").map(&:safe_constantize)
+      klasses.inject(block) { |inner, klass| -> { klass.with_deleted(&inner) } }.call
+    else
+      yield
     end
   end
 
-  # For both deleted and non-deleted resources
-  def with_deleted?
-    params[:with_deleted] == "true"
-  end
-
   def search_with_deleted?
-    params.dig(:search, :deleted) == "true"
+    params.dig(:search, :deleted)
```

Comment:
Need to change the view to pass "Project,Stage,Deploy" for `search[:deleted]` field or else the value will be `nil` when we try to filter in deploys page.

Reply:
ah yes, good catch!
I'm setting with_deleted from search query now to avoid having to check multiple params and so links work